src/org/traccar/protocol/Ardi01ProtocolDecoder.java

```diff
@@ -75,12 +75,16 @@ protected Object decode(
         position.setLatitude(parser.nextDouble());
         position.setSpeed(UnitsConverter.knotsFromKph(parser.nextDouble()));
         position.setCourse(parser.nextDouble());
-        position.setAltitude(parser.nextDouble());
+        double altitude = parser.nextDouble();
 
         int satellites = parser.nextInt();
         position.setValid(satellites >= 3);
         position.set(Position.KEY_SATELLITES, satellites);
 
+        if (satellites >= 4) {
```

Comment:
To calculate a location in two dimensions (i.e., lat, lon), you need at least three satellites in the fix.
To calculate a location in three dimensions (i.e., lat, lon, alt), you need at least four satellites in the fix.